src/test/ui/consts/const-float-classify.rs

```diff
@@ -0,0 +1,63 @@
+// -Zmir-opt-level=0
+// run-pass
+
+#![feature(const_panic)]
+#![feature(const_if_match)]
+#![feature(const_float_bits_conv)]
+#![feature(const_float_classify)]
+
+// Don't promote
+const fn nop<T>(x: T) -> T { x }
+
+macro_rules! const_assert {
+    ($a:expr, $b:expr) => {
+        {
+            const _: () = assert!($a == $b);
+            assert_eq!(nop($a), nop($b));
+        }
+    };
+}
+
+macro_rules! suite {
+    ( $( $tt:tt )* ) => {
+        fn f32() {
+            suite_inner!(f32 $($tt)*);
+        }
+
+        fn f64() {
+            suite_inner!(f64 $($tt)*);
+        }
+    }
+
+}
+
+macro_rules! suite_inner {
+    (
+        $ty:ident [$( $fn:ident ),*]
+        $val:expr => [$($out:ident),*]
+
+        $( $tail:tt )*
+    ) => {
+        $( const_assert!($ty::$fn($val), $out); )*
+        suite_inner!($ty [$($fn),*] $($tail)*)
+    };
+
+    ( $ty:ident [$( $fn:ident ),*]) => {};
+}
+
+suite! {
+                   [is_nan, is_infinite, is_finite, is_normal, is_sign_positive, is_sign_negative]
+     -0.0 / 0.0 => [  true,       false,     false,     false,             true,            false]
+      0.0 / 0.0 => [  true,       false,     false,     false,             true,            false]
+            1.0 => [ false,       false,      true,      true,             true,            false]
+           -1.0 => [ false,       false,      true,      true,            false,             true]
+            0.0 => [ false,       false,      true,     false,             true,            false]
+           -0.0 => [ false,       false,      true,     false,            false,             true]
+      1.0 / 0.0 => [ false,        true,     false,     false,             true,            false]
+     -1.0 / 0.0 => [ false,        true,     false,     false,            false,             true]
```

Comment:
Would it make sense to also add the `fX::{NAN,INFINITY,NEG_INFINITY}` constants to this table?

Reply:
If you figure out how to add that to the macro then sure. Since they're defined as `0 / 0`, `1 / 0` and `-1 / 0` respectively I don't think it's important.

Reply:
Oh, one interesting note is that `-0.0 / 0.0` happens to have a positive sign both on Miri and in x86_64 release mode. This is not guaranteed anywhere, and it might differ between platforms (especially in debug mode where const propagation doesn't apply).

Reply:
AFAIK mir-opt-level=1 is used even for debug builds?

Reply:
> If you figure out how to add that to the macro then sure. Since they're defined as 0 / 0, 1 / 0 and -1 / 0 respectively I don't think it's important.

Fair.

Reply:
This is LLVMs const propagation. I think I need to rewrite this test to not assume anything about NaN signedness, since it's likely to fail on other platforms.